server/storage/interface.go

```diff
@@ -1,14 +1,31 @@
 package storage
 
-// KeyStore provides a minimal interface for managing key persistence
-type KeyStore interface {
-	// GetKey returns the algorithm and public key for the given GUN and role.
-	// If the GUN+role don't exist, returns an error.
-	GetKey(gun, role string) (algorithm string, public []byte, err error)
+import (
+	"time"
+
+	"github.com/docker/notary/tuf/data"
+)
 
-	// SetKey sets the algorithm and public key for the given GUN and role if
-	// it doesn't already exist.  Otherwise an error is returned.
-	SetKey(gun, role, algorithm string, public []byte) error
+// KeyStore provides a minimal interface for managing key persistence.  Any
+// or all of these functions can optionally clean up expired pending keys, or
+// no-longer-active non-pending keys.
+type KeyStore interface {
+	// GetLatestKey the most recently created, non-expired key for the given
+	// GUN and role. If no keys exist for the GUN+role, returns an ErrNoKeys.
+	GetLatestKey(gun, role string) (*ManagedPublicKey, error)
+
+	// HasAnyKeys returns true if any non-expired keys exist for the given GUN,
```

Comment:
Only necessary because we validate that the root.json has timestamp keys that the server has. If we want to eliminate this requirement and allow users to manage their own timestamp keys, we can probably eliminate this interface.

It is also used as a shortcut to see if we can generate a snapshot, but we can probably just try to generate one and then fail if we don't have the keys.